### PR TITLE
Add epoch query API to authorityAPI

### PR DESCRIPTION
**Pull request by lxfind**
Add an API to AuthorityAPI that can request authenticated epoch data structure.
Cannot add the proper verify function in SafeClient yet, because that depends on hooking up the SafeClient and the state store in order to fetch the committee.

### CODE REVIEW

File: crates/sui-core/src/safe_client.rs [hunk 1]
```diff
@@ -581,4 +582,56 @@ where
...
+        let authority = self.address;
+        let response = self.authority_client.handle_epoch(request).await?;
+        self.verify_epoch(epoch_id, &response)
+            .map_err(|err| SuiError::ByzantineAuthoritySuspicion {
+                authority,
+                reason: err.to_string(),
+            })
+            .tap_err(|err| {
+                self.report_client_error(err);
+            })?;
```
Comment (bmwill) [inline crates/sui-core/src/safe_client.rs:634]: Generally this is not how logging should be handled, the macro invocation should be inlined here in order to properly report line numbers, etc.

**Review by bmwill — APPROVED** [review]
lgtm aside from one nit

File: crates/sui-core/src/safe_client.rs (hunk 1 above)
Comment (lxfind) [reply crates/sui-core/src/safe_client.rs:634]: Good point

File: crates/sui-core/src/authority_active/gossip/configurable_batch_action_client.rs [hunk 2]
```diff
@@ -205,6 +205,10 @@ impl AuthorityAPI for ConfigurableBatchActionClient {
     ) -> Result<CheckpointResponse, SuiError> {
         todo!();
     }
+
+    async fn handle_epoch(&self, _request: EpochRequest) -> Result<EpochResponse, SuiError> {
+        todo!();
```
Comment (asonnino) [inline crates/sui-core/src/authority_active/gossip/configurable_batch_action_client.rs:210]: Do we have an issue to track this?

**Review by asonnino — APPROVED** [review]

File: crates/sui-core/src/safe_client.rs (hunk 1 above)
Comment (lxfind) [reply crates/sui-core/src/safe_client.rs:634]: Filed https://github.com/MystenLabs/sui/issues/3740

**Pull request merged**